examples/migration/src/ledger.rs

```diff
@@ -222,34 +222,60 @@ struct BesuContractConfig {
 }
 
 impl BesuLedger {
-    fn contracts(contracts: &BesuContracts) -> Vec<ContractConfig> {
-        vec![
-            ContractConfig {
-                address: contracts.did_registry.address.to_string(),
-                spec_path: Some(contracts.did_registry.path.to_string()),
-                spec: None,
-            },
-            ContractConfig {
-                address: contracts.schema_registry.address.to_string(),
-                spec_path: Some(contracts.schema_registry.path.to_string()),
-                spec: None,
-            },
-            ContractConfig {
-                address: contracts.credential_definition_registry.address.to_string(),
-                spec_path: Some(contracts.credential_definition_registry.path.to_string()),
-                spec: None,
-            },
-            ContractConfig {
-                address: contracts.role_control.address.to_string(),
-                spec_path: Some(contracts.role_control.path.to_string()),
-                spec: None,
-            },
-            ContractConfig {
-                address: contracts.legacy_mapping_registry.address.to_string(),
-                spec_path: Some(contracts.legacy_mapping_registry.path.to_string()),
-                spec: None,
-            },
-        ]
+
+    pub const CHAIN_ID: u64 = 1337;
+    pub const NODE_ADDRESS: &'static str = "http://127.0.0.1:8545";
+
+    fn build_contract_path(contract_path: &str) -> String {
+        let mut cur_dir = env::current_dir().unwrap();
+        cur_dir.push("../../"); // project root directory
+        cur_dir.push(contract_path);
+        fs::canonicalize(&cur_dir)
+            .unwrap()
+            .to_str()
+            .unwrap()
+            .to_string()
+    }
+
+    fn contracts() -> Vec<ContractConfig> {
```

Comment:
I think it makes sense to add explicit structure definition for the config file.
The function of reading the config will be much easier.

```
#[derive(Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
struct BesuContracts {
    did_registry: BesuContractConfig,
    schema_registry: BesuContractConfig,
    credential_definition_registry: BesuContractConfig,
    role_control: BesuContractConfig,
    legacy_mapping_registry: BesuContractConfig,
    .......
}
```

Reply:
done